compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/TypeAliasAnnotationMethodsLowering.kt

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2010-2019 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.backend.jvm.lower
+
+import org.jetbrains.kotlin.backend.common.CommonBackendContext
+import org.jetbrains.kotlin.backend.common.FileLoweringPass
+import org.jetbrains.kotlin.backend.common.descriptors.WrappedSimpleFunctionDescriptor
+import org.jetbrains.kotlin.backend.common.phaser.makeIrFilePhase
+import org.jetbrains.kotlin.descriptors.Modality
+import org.jetbrains.kotlin.ir.IrElement
+import org.jetbrains.kotlin.ir.UNDEFINED_OFFSET
+import org.jetbrains.kotlin.ir.builders.declarations.addFunction
+import org.jetbrains.kotlin.ir.builders.declarations.buildFun
+import org.jetbrains.kotlin.ir.declarations.*
+import org.jetbrains.kotlin.ir.declarations.impl.IrFunctionImpl
+import org.jetbrains.kotlin.ir.expressions.impl.IrBlockBodyImpl
+import org.jetbrains.kotlin.ir.symbols.impl.IrSimpleFunctionSymbolImpl
+import org.jetbrains.kotlin.ir.visitors.IrElementVisitorVoid
+import org.jetbrains.kotlin.ir.visitors.acceptChildrenVoid
+import org.jetbrains.kotlin.ir.visitors.acceptVoid
+import org.jetbrains.kotlin.load.java.JvmAbi
+import org.jetbrains.kotlin.name.Name
+
+
+internal val typeAliasAnnotationMethodsPhase = makeIrFilePhase(
+    ::TypeAliasAnnotationMethodsLowering,
+    name = "TypeAliasAnnotationMethodsLowering",
+    description = "Generate method stubs for type alias annotations"
+)
+
+class TypeAliasAnnotationMethodsLowering(val context: CommonBackendContext) :
+    IrElementVisitorVoid,
+    FileLoweringPass {
+
+    override fun lower(irFile: IrFile) {
+        irFile.acceptVoid(this)
+    }
+
+    override fun visitElement(element: IrElement) {
+        element.acceptChildrenVoid(this)
+    }
+
+    override fun visitFile(declaration: IrFile) {
```

Comment:
This override is redundant cause of fileClassPhase, that moves file content into class

Reply:
...and lower could be switched to ClassLoweringPass

Reply:
Is the same true of PropertiesLowering?

Reply:
It is! Just tried tried out the refactoring of PropertiesLowering to a ClassLoweringPass. Do you want that in this PR, or should I extract that to a separate one?

Reply:
It's best to make separate ibe

Reply:
Oh, no. PropertiesLowering is common lowering. And as i remeber Native and JS don't perform fileClassPhase

Reply:
Ah, right! I just spotted the same as I went to check the changes in and I noticed the path was different.